### PR TITLE
Bug fix in RoundFloat func in common/math package 

### PR DESCRIPTION
**Pull request by vazha**
# PR Description

RoundFloat refactored, TestRoundFloat refactored and added more test cases

Fixes # (issue)

RoundFloat function did not work correctly. 
For example if we test value 2.3582323 on precision 2 we get  2.37 instead of 2.36

## Type of change

- [v ] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

go test

## Checklist

- [v] My code follows the style guidelines of this project
- [v] I have performed a self-review of my own code
- [v] My changes generate no new warnings

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -357,6 +357,7 @@ github.com/urfave/cli v1.22.4 h1:u7tSpNPPswAFymm8IehJhy4uJMlUuU/GmqSkvJ1InXA=
 github.com/urfave/cli v1.22.4/go.mod h1:Gos4lmkARVdJ6EkW0WaNv/tZAAMe9V7XWyB60NtXRu0=
 github.com/urfave/cli/v2 v2.2.0 h1:JTTnM6wKzdA0Jqodd966MVj4vWbbquZykeX1sKbe2C4=
 github.com/urfave/cli/v2 v2.2.0/go.mod h1:SE9GqnLQmjVa0iPEY0f1w3ygNIYcIJ0OKPMoW2caLfQ=
+github.com/vazha/gocryptotrader v0.1.4 h1:Qb4qpAKn2/N5p2bU2vN8YuGEimUTHXztot0kzOqj2lo=
```
Comment (thrasher-) [inline go.sum:360]: Just need to remove this and the rest looks good!

**Review by thrasher- — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (vazha) [reply go.sum]: done :)

**Review by xtda — APPROVED** [review]
tested approved !

**Review by gloriousCode — APPROVED** [review]
tACK! 🎉  thanks for your contribution @vazha

**Review by shazbert — APPROVED** [review]
tACK 👍

**Review by thrasher- — APPROVED** [review]

**Pull request merged**